cpp/terrain/terrain_object.h

```diff
@@ -108,12 +108,15 @@ class TerrainObject : public std::enable_shared_from_this<TerrainObject> {
 	bool is_floating() const;
 
 	/**
-	 * has the object been placed
+	 * returns true if this object has been placed. this indicates that the object has a position and exists
+	 * on the map, floating buildings are not considered placed as they are only an indicator
+	 * for where something can begin construction
```

Comment:
there is an enum, but there is more than one state that will make those function true.